src/splitter.rs

```diff
@@ -50,4 +153,195 @@ mod tests {
+        let splitter = Splitter::new();
+        let text = "hello world";
+        let phrases = splitter.phrases(text);
+        
+        // Should have one phrase of length 2: ["hello", "world"]
+        assert_eq!(phrases.len(), 1);
+        assert_eq!(phrases[0].len(), 2);
+    }
+
+    #[test]
+    fn test_phrases_no_single_words() {
+        let splitter = Splitter::new();
+        let text = "word";
+        let phrases = splitter.phrases(text);
+        
+        // Single word should produce no phrases
+        assert!(phrases.is_empty());
+    }
+
+    #[test]
+    fn test_phrases_substrings() {
+        let splitter = Splitter::new();
+        let text = "one two three four";
+        let phrases = splitter.phrases(text);
+        
+        // Should have 6 phrases: 1 of length 4, 2 of length 3, 3 of length 2
+        assert_eq!(phrases.len(), 6);
+        
+        // Check we have different lengths
+        let mut lengths: Vec<usize> = phrases.iter().map(|p| p.len()).collect();
+        lengths.sort();
+        assert_eq!(lengths, vec![2, 2, 2, 3, 3, 4]);
+    }
+
+    #[test]
+    fn test_phrases_sentence_delimiters() {
+        let splitter = Splitter::new();
+        let text = "hello world. foo bar?";
+        let phrases = splitter.phrases(text);
+        
+        // Should have phrases from both sentences
+        // "hello world" -> 1 phrase
+        // "foo bar" -> 1 phrase
+        // Total: 2 phrases (assuming no duplicates)
+        assert_eq!(phrases.len(), 2);
+    }
+
+    #[test]
+    fn test_phrases_punctuation_removal() {
+        let splitter = Splitter::new();
+        let text = "hello, world! it's working.";
+        let phrases = splitter.phrases(text);
+        
+        // Should clean punctuation but keep 's
+        // Should have phrases with cleaned words
+        assert!(!phrases.is_empty());
+    }
+
+    #[test]
+    fn test_phrases_vocabulary_indexing() {
+        let splitter = Splitter::new();
+        let text = "apple banana apple";
+        let phrases = splitter.phrases(text);
+        
+        // Vocabulary should be ["apple", "banana"] (sorted)
+        // Phrases should reference indices correctly
+        // For "apple banana apple" we expect:
+        // - "apple banana" (length 2)
+        // - "banana apple" (length 2) 
+        // - "apple banana apple" (length 3)
+        assert_eq!(phrases.len(), 3); // Updated expectation
+        
+        // Check that indices are valid u16 values
+        for phrase in &phrases {
+            for &index in phrase {
+                assert!(index < 2); // Only 2 words in vocabulary
+            }
+        }
+    }
+
+    #[test] 
+    fn test_phrases_no_duplicates() {
+        let splitter = Splitter::new();
+        let text = "hello world. hello world?";
+        let phrases = splitter.phrases(text);
+        
+        // Both sentences produce the same phrase, should only appear once
+        assert_eq!(phrases.len(), 1);
```

Comment:
Make this a little stronger - it should remove duplicates within and across sentences.